addons/account/models/company_onboarding.py

```diff
@@ -0,0 +1,161 @@
+                'user_type_id': sample_user_type_id.id,
+                'reconcile': True,
+                'company_id': self.env.user.company_id.id,
+            })
+
+        return sample_account
+
+    def _get_sample_invoice(self):
+        """ Get a sample invoice or create it as well as a sample account,
+           journal, partner and line if it does not exist. """
+
+        # try to use a partner that has the same email as the current user
+        # instead of creating a new one
+        sample_partner = self.env['res.partner'].search([('email', '=', self.env.user.email)], limit=1)
+
+        if len(sample_partner) == 0:
+            sample_partner_name = _('Sample Partner')
+            sample_partner = self.env['res.partner'].search([('name', '=', sample_partner_name)], limit=1)
+
+            account_other_id = self._get_sample_account('other', 'SACOO').id
+            if len(sample_partner) == 0:
+                sample_partner = self.env['res.partner'].create({
+                    'name': sample_partner_name,
+                    'customer': True,
+                    'email': self.env.user.email,
+                    'property_account_payable_id': account_other_id,
+                    'property_account_receivable_id': account_other_id,
+                })
+
+        sample_journal_code = 'SAJO'
+        sample_journal = self.env['account.journal'].search([('code', '=', sample_journal_code)], limit=1)
+        if len(sample_journal) == 0:
+            sample_journal = self.env['account.journal'].create({
+                'name': _('Sample Journal'),
+                'code': sample_journal_code,
+                'type': 'sale',
+            })
+
+        sample_invoice_name = self.env['account.invoice'].SAMPLE_INVOICE_NAME
+        sample_invoice = self.env['account.invoice'].search([('name', '=', sample_invoice_name)], limit=1)
+        if len(sample_invoice) == 0:
+            sample_invoice = self.env['account.invoice'].create({
+                'name': sample_invoice_name,
+                'journal_id': sample_journal.id,
+                'partner_id': sample_partner.id,
+            })
+
+            sample_line = self.env['account.invoice.line'].create({
+                'name': _('Sample line'),
+                'invoice_id': sample_invoice.id,
+                'account_id': self._get_sample_account('receivable', 'SACOR').id,
+                'price_unit': 199.99,
+                'quantity': 2,
+            })
+
+        return sample_invoice
+
+    @api.model
+    def onboarding_sample_invoice(self, company_id):
+        """ Open a window to compose an email, with the edi_invoice_template
+            message loaded by default. """
+        sample_invoice = self._get_sample_invoice()
+        template = self.env.ref('account.email_template_edi_invoice', False)
+        compose_form = self.env.ref(
+            'account.email_compose_onboarding_sample_invoice', False)
+        ctx = dict(
```

Comment:
style: why 2 different literra for dict, and what about defining the action in xml ?